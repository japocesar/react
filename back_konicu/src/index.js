import React from 'react';
import ReactDOM from 'react-dom';
import { KonicuApp } from './KonicuApp';


ReactDOM.render(
  
    <KonicuApp />,

  document.getElementById('root')
);