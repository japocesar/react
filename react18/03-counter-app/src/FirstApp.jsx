import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name } ) => {
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    )
}


FirstApp.propTypes = {
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Cesar Japo',
    subtitle: 'No Subtitle',
    // title: 'No title',
}