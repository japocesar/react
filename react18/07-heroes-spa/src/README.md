## Installation

Install the dependencies and devDependencies and start the server.

```sh
yarn
yarn dev
```

Open the "Local" address on your browser