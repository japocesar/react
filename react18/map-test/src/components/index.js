export * from './MapComponent';
export * from './ScoreComponent';