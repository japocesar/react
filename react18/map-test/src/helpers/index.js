export * from './capitalCities';
export * from './mapConfig';
export * from './distanceCalculator';