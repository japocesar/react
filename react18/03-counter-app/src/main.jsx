import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';
// import HelloWorldApp from './HelloWorld';


ReactDOM.createRoot( document.querySelector("#root") ).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirstApp title="I'm Batman" /> */}
        { <CounterApp value={ 10 } /> }
    </React.StrictMode>
)
