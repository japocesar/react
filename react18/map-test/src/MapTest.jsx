import { useState } from "react";
import { MapComponent, ScoreComponent } from "./components";

import { capitalCities } from './helpers';

export const MapTest = () => {

  const [cityIndex, setCityIndex] = useState(0);
  const [cityName, setCityName] = useState(capitalCities[0].capitalCity);
  const [latLng, setLatLng] = useState({ lat: capitalCities[0].lat, lng: capitalCities[0].long });
  const [score, setScore] = useState(1500);
  const [cityCounter, setCityCounter] = useState(0);
  const [gameOver, setGameOver] = useState(false);

  const onUpdateIndex = ( distance ) => {
    try {
      if ( distance > 50 ) {
        const currentScore = (score - distance).toFixed(2);
        setScore( currentScore );
        if( currentScore <= 0 ) {
          setGameOver(true);
          setScore(0);
        }
      } else {
        setCityCounter( cityCounter + 1 );
      }

      setCityIndex(cityIndex+1);
      setCityName( capitalCities[cityIndex+1].capitalCity );
      setLatLng({ lat: capitalCities[cityIndex+1].lat, lng: capitalCities[cityIndex+1].long });
    }
    catch (e) {
      setGameOver(true);
    }
  }

  return (
    <div className="container">
        <h1>MapTest</h1>
        <hr />
        <div className="row">
            <div className="col-12 col-md-4">
                <ScoreComponent cityIndex={ cityIndex } score={ score } cityCounter={cityCounter} gameOver={ gameOver }/>
            </div>
            <div className="col-12 col-md-8 mt-5 mt-md-0">
                <MapComponent cityName={cityName} cityLatLng={ latLng } onPlaceMarker={ onUpdateIndex }  gameOver={ gameOver }   />
            </div>
        </div>
    </div>
  )
}
