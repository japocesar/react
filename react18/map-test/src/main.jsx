import React from 'react';
import ReactDOM from 'react-dom/client';
import { MapTest } from './MapTest';
import './custom.scss';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <MapTest />
  </React.StrictMode>
)
