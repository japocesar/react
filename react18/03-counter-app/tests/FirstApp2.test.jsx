import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Tests in FirstApp', () => {
    const title = "I'm Batman";
    const subtitle = 'This is a subtitle';

    test('should match with snapshot', () => { 
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();
    })

    test('should display the message "Im Batman"', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
        // screen.debug();
    })

    test('should display the title in an h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    })

    test('should display the subtitle by props', () => { 
        render( 
            <FirstApp 
                title={ title } 
                subtitle= { subtitle }    
            /> 
        );
        expect( screen.getAllByText( subtitle ).length ).toBe(1);
     })
})