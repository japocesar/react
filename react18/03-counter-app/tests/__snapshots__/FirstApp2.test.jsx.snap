// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    I'm Batman
  </h1>
  <p>
    No Subtitle
  </p>
  <p>
    Cesar Japo
  </p>
</div>
`;
